### ISSUE TITLE
Creating BGZFile results in TypeError

### ISSUE THREAD

**Issue by p-himik** [pysam-developers/pysam#411]
```python
In [67]: pysam.BGZFile(name)
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-67-f1b1b09e86f7> in <module>()
----> 1 pysam.BGZFile(name)

pysam/libcbgzf.pyx in pysam.libcbgzf.BGZFile.__init__ (pysam/libcbgzf.c:1804)()

TypeError: expected bytes, str found

In [68]: pysam.__version__
Out[68]: '0.10.0'
```

**Comment by bioinformed** [comment]
Thanks for the bug report.  I'll take a look.

**Comment by bioinformed** [comment]
Fixed with bdb746ca441123ec0b8c7143d8bdc38ab0d6e4e1
